pyiron_base/storage/has_stored_traits.py

```diff
@@ -309,6 +309,7 @@ def lock(self):
         """Recursively make all traits read-only."""
         self._read_only = True
         for sub in self.trait_values().values():
+            if not isinstance(sub, HasStoredTraits): continue
             try:
                 sub.lock()
             except AttributeError:
```

Comment:
I thought we no longer need the duck typing at all? (Also the suggestion is incomplete because GitHub won't let me touch the `pass` line at the end...)

```suggestion
            if isinstance(sub, HasStoredTraits):
                sub.lock()
```

Reply:
Added a commit to the same effect.